### PR TITLE
Interfaces for working with abstract properties of Collections and Documents

### PR DESCRIPTION
**Pull request by adamretter**

### CODE REVIEW

File: src/org/exist/ResourceMetadata.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+public interface ResourceMetadata {
+
+    /**
+     * Get the creation time of the resource
+     *
+     * @return the difference, measured in milliseconds, between the time when
+     * the resource was created and midnight, January 1, 1970 UTC.
+     */
+    long getCreated();
```
Comment (shabanovd) [inline src/org/exist/ResourceMetadata.java]: where `getMimeType`, `getLastModified` methods?

File: src/org/exist/collections/Collection.java [hunk 2]
```diff
@@ -83,7 +84,7 @@
  *
  * @author wolf
  */
-public class Collection extends Observable implements Comparable<Collection>, Cacheable {
+public class Collection extends Observable implements Resource, Comparable<Collection>, Cacheable {
```
Comment (shabanovd) [inline src/org/exist/collections/Collection.java]: `getCreated`, `getMimeType`, `getLastModified` should be `@deprecated`

File: src/org/exist/ResourceMetadata.java (hunk 1 above)
Comment (adamretter) [inline src/org/exist/ResourceMetadata.java]: Collections have never had a `lastModified` time or `mimeType`

File: src/org/exist/collections/Collection.java (hunk 2 above)
Comment (adamretter) [inline src/org/exist/collections/Collection.java]: Do you mean in favour of the ones in `CollectionMetadata`?

Comment (shabanovd) [inline src/org/exist/collections/Collection.java]: yes

Comment (adamretter) [inline src/org/exist/collections/Collection.java]: Fixed!

**Pull request merged**